python/dgl/sampling/pinsage.py

```diff
@@ -0,0 +1,214 @@
+"""PinSAGE sampler & related functions and classes"""
+
+import numpy as np
+
+from .. import backend as F
+from .. import convert
+from .. import transform
+from .. import function as fn
+from .randomwalks import random_walk
+from .neighbor import select_topk
+from ..base import EID
+
```

Comment:
Add a `__all__`? You use `from pinsage import *` and I suppose the `GenericPinSAGESampler` is an internal class. Do we have other classes wish to inherit this generic class?

Reply:
I actually don't want it to be an internal class.  For instance, one can also use `GenericPinSAGESampler` on a homogeneous graph.  `PinSAGESampler` is just a convenient wrapper on `GenericPinSAGESampler` for bidirectional bipartite graphs.

Reply:
Can we extend `PinSAGESampler` to make it work directly for homograph? I guess adding some default argument values might do the trick.

Reply:
In fact any heterographs can have a PinSAGE sampler running given a metapath that has the same start and end node type.